tests/tests/test_mender_gateway.py

```diff
@@ -52,6 +108,52 @@ def do_test_deployment_one_device(self, env, valid_image_with_mender_conf):
             devices=[device_id],
         )
 
+    def do_test_deployment_gateway_and_one_device(
+        self,
+        env,
+        valid_image_with_mender_conf,
+        image_with_mender_conf_and_mender_gateway_conf,
+    ):
+        mender_device = env.device
+        mender_gateway = env.device_gateway
+        devauth = DeviceAuthV2(env.auth)
+        deploy = Deployments(env.auth, devauth)
+
+        host_ip = env.get_virtual_network_host_ip()
+
+        gateway_id = Helpers.ip_to_device_id_map(
+            MenderDeviceGroup([mender_gateway.host_string]), devauth=devauth,
+        )[mender_gateway.host_string]
+
+        mender_conf = mender_gateway.run("cat /etc/mender/mender.conf")
+        mender_gateway_conf = mender_gateway.run("cat /etc/mender/mender-gateway.conf")
+        mender_gateway_image = image_with_mender_conf_and_mender_gateway_conf(
+            "mender-gateway-image-full-cmdline-%s.ext4" % conftest.machine_name,
+            mender_conf,
+            mender_gateway_conf,
+        )
+
+        deployment_id, _ = common_update_procedure(
+            mender_gateway_image, devices=[gateway_id], devauth=devauth, deploy=deploy,
+        )
+
+        device_id = Helpers.ip_to_device_id_map(
+            MenderDeviceGroup([mender_device.host_string]), devauth=devauth,
+        )[mender_device.host_string]
+
+        mender_conf = mender_device.run("cat /etc/mender/mender.conf")
+        update_image(
+            mender_device,
+            host_ip,
+            expected_mender_clients=1,
+            install_image=valid_image_with_mender_conf(mender_conf),
+            devauth=devauth,
+            deploy=deploy,
+            devices=[device_id],
+        )
+
+        deploy.check_expected_statistics(deployment_id, "success", 1)
```

Comment:
This should be fine, but I would minimize the chances of one deployment finishing before the other start. Doing tests time based like this is always a bit risky.

Three things to consider:
* `ip_to_device_id_map` uses SSH commands, which take a while, so I would do this before `common_update_procedure( mender_gateway_image ...`
* Same for `mender_device.run("cat /etc/mender/mender.conf")`
* Lastly, consider also using `deployment_triggered_callback` from `common_update_procedure` to trigger the second deployment, so that `common_update_procedure` does not even wait for the deployment to start (see that otherwise it wait until the deployment moves from `pending` to other state).

Reply:
Done here: https://github.com/mendersoftware/integration/pull/1923/commits/8a155eaba7678c741bd72e0fe1b214a6f24e886a